src/main.rs

```diff
@@ -269,12 +294,11 @@ async fn main() {
 
     let ts = Arc::new(TMutex::new(Instant::now()));
     let mut handles = Vec::new();
+    // For the sake of efficient memory usage 'UpdateState' is created only once
+    // and then passed as a reference.
```

Comment:
Nitpick, but it seems a bit weird to call `Arc` a "reference". On a very technical level, it basically is, but when I think "reference" in Rust, I think `&`, not `Arc`. Perhaps just lose the second line entirely?

```suggestion
```